src/util.h

```diff
@@ -69,15 +70,50 @@ inline std::string _(const char* psz)
 void SetupEnvironment();
 bool SetupNetworking();
 
+enum LogFlags : uint32_t {
+    LOG_NONE        = 0,
+    LOG_NET         = (1 <<  0),
```

Comment:
19:32 < gmaxwell> wumpus: uh. so boolean operations don't appear to exist for scoped enums. so ... I don't think I can use them without 
                  defining a set of boolean operations for them. 
19:32 < sipa> gmaxwell: eh?
19:33 < sipa> ah, scoped enums don't have an implicit conversion to int
19:34 < sipa> you can still explicitly convert them, though
19:34 < sipa> but enums are generally a bad approach for bitfields, as they're intended to be exhaustive
19:34 < gmaxwell> sipa: wumpus suggested that I change the log print thing to use a scoped enum.  The original thing that inspired this 
                  change is that rebroad wanted to do some horrible hack to do the logical equivilent of LogPrint(LOG_NET | LOG_MEMPOOL, 
                  ...). (and of course the matching also uses it as a bitfield.)
19:35 < sipa> right, a normal enum (which behaves much more like an int) may be more appropriate
19:35 < gmaxwell> well thats what I implemented.
19:37 < sipa> awesöme
19:37 < gmaxwell> in any case. I'm happy to change it to whatever. someone who has an opinion needs to tell me what to do. I think I only 
                  need a couple casts to make the enum class stuff go, but it would be ugly to do "LogPrint(LOG_NET | LOG_MEMPOOL," later 
                  unless we provide | for the type.